STYLE.md

```diff
@@ -0,0 +1,631 @@
+
+- Avoid: `basicAuthUsername, basicAuthPassword, _ := r.BasicAuth()`
+- Prefer: `username, password, _ := r.BasicAuth()`
+
+In context, it's obvious that the username and password are for basic auth.
+
+### Be consistent
+
+If `foo()` returns a "code", then `code := foo()` is more accurate and more concise than `returnVal := foo()` -- the result variable should be named consistently with the function called.
+
+If the rest of the API uses `verbNoun` then unless there is a very good reason not to, the next function should be of the form `verbNoun`. For example, if the API has `createUser()`, `updateUser()`, `deleteUser()`, then adding `getUser()` fits perfectly, but if you add `userFetch()` instead of `fetchUser()`, it breaks the consistent pattern, making the API harder to learn and use.
+
+## Code style
+
+### Merge arguments of the same type
+
+- Avoid: `func foo(a string, b string, c string)`
+- Prefer: `func foo(a, b, c string)`
+
+### Named return values
+
+It's usually clearer to use named arguments when you're returning multiple values (apart from `error`), for example:
+
+- Avoid: `func run() (<-chan servicelog.Entry, <-chan servicelog.Entry) {}`
+- Prefer: `func run() (stdoutCh, stderrCh <-chan servicelog.Entry) {}`
+
+### Short variable declarations
+
+Where possible, use the more idiomatic `:=` short form variable declaration with an implicit type.
+
+- Avoid: `var a = 0`
+- Prefer: `a := 0`, even if you must explicitly assign a zero value. This will help signify that there are code-paths which read this value before it is assigned to.
+
+On the other hand, if the zero value would never be read, use `var foo type`. For example:
+
+```go
+var name string
+if i == 1 {
+	name = "one"
+} else {
+	name = "not one"
+}
+```
+
+That said, in cases like this is sometimes simpler to initialise to the `else` value instead:
+
+```go
+name := "not one"
+if i == 1 {
+    name = "one"
+}
+```
+
+[//]: # (### Avoid very small functions)
+
+[//]: # ()
+[//]: # (In many cases, it's not worth creating a function or method for one-liners that aren't exported. For example, below you could simply write `positions` inline:)
+
+[//]: # ()
+[//]: # (Avoid:)
+
+[//]: # ()
+[//]: # (```go)
+
+[//]: # (func &#40;rb *RingBuffer&#41; Positions&#40;&#41; &#40;start RingPos, end RingPos&#41; {)
+
+[//]: # (	rb.rwlock.RLock&#40;&#41;)
+
+[//]: # (	defer rb.rwlock.RUnlock&#40;&#41;)
+
+[//]: # (	return rb.positions&#40;&#41;)
+
+[//]: # (})
+
+[//]: # ()
+[//]: # (func &#40;rb *RingBuffer&#41; positions&#40;&#41; &#40;start RingPos, end RingPos&#41; {)
+
+[//]: # (	return rb.readIndex, rb.writeIndex)
+
+[//]: # (})
+
+[//]: # (```)
+
+[//]: # ()
+[//]: # (Prefer:)
+
+[//]: # ()
+[//]: # (```go)
+
+[//]: # (func &#40;rb *RingBuffer&#41; Positions&#40;&#41; &#40;start RingPos, end RingPos&#41; {)
+
+[//]: # (	rb.rwlock.RLock&#40;&#41;)
+
+[//]: # (	defer rb.rwlock.RUnlock&#40;&#41;)
+
+[//]: # (	return rb.readIndex, rb.writeIndex)
+
+[//]: # (})
+
+[//]: # (```)
```

Comment:
I don't agree with this. I think the example they mention makes sense, but there are cases where one liners make sense, e.g. instead of
```
func f(user: User) {
    // We consider males that are not adults as boys
     if (user.sex == "m" && user.age < 18:)
         # do logic
}
```
I would write

```
func f(user User) {
      if (isBoy(user)):
           # do logic
}

func isBoy(user User {
      return user.sex == "m" && user.age < 18
}

Reply:
Yes, isolated testing is another benefit. My main focus with the example was more on readability though.